fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java

```diff
@@ -139,13 +145,24 @@ public void open() throws IOException {
             return;
         }
         try {
-            TableTunnel.DownloadSession session = curTableScan.getSession();
+            TableTunnel.DownloadSession session;
+            if (partitionSpec != null) {
+                PartitionSpec partSpec = new PartitionSpec(partitionSpec);
+                session = curTableScan.openDownLoadSession(partSpec);
+            } else {
+                session = curTableScan.openDownLoadSession();
+            }
             long start = startOffset == -1L ? 0 : startOffset;
             long recordCount = session.getRecordCount();
             totalRows = splitSize > 0 ? Math.min(splitSize, recordCount) : recordCount;
 
             arrowAllocator = new RootAllocator(Long.MAX_VALUE);
-            curReader = session.openArrowRecordReader(start, totalRows, readColumns, arrowAllocator);
+            Set<String> partitionColumns = session.getSchema().getPartitionColumns().stream()
+                    .map(Column::getName)
+                    .collect(Collectors.toSet());
+            List<Column> maxComputeColumns = new ArrayList<>(readColumns);
+            maxComputeColumns.removeIf(e -> partitionColumns.contains(e.getName()));
```

Comment:
Why remove partition columns here?

Reply:
max compute depart the partition columns from common columns